imports/plugins/included/product-detail-simple/client/components/productDetail.js

```diff
@@ -74,6 +94,117 @@ class ProductDetail extends Component {
     return null;
   }
 
+  switchDigital(e) {
+    this.setState({digital: e.target.checked});
+    const productId = this.props.product._id;
+    const checked = e.target.checked;
+    this.props.onProductFieldChange(productId, "isDigital", checked);
+    this.props.changeParentIsDigitalState(e.target.checked);
+    if (!this.state.digital) {
+      this.setState({cartQuantityType: "hidden"});
+    } else {
+      this.setState({cartQuantityType: "number"});
+    }
+  }
+
+  renderInputToggle() {
+    if (this.props.hasAdminPermission) {
+      return (
+        <div className="checkbox checkbox-primary">
+          <input id="isdigital" type="checkbox" defaultChecked={this.state.digital} onChange={this.switchDigital}/>
+          <label for="isdigital">Product is Digital </label>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  handleCategoryStateChange(e) {
+    const value = e.target.value;
+    this.setState({categoryValue: value});
+  }
+
+  handleUploadClick(e) {
+    const files = new FS.File(e.target.files[0]);
+    files.metadata = {
+    productId: this.props.product._id,
+    ownerId: Meteor.userId(),
+    shopId: Reaction.getShopId()
+    };
+
+    const insertDigitalFile = (db, Alerts) => {
+      db.insert(files, (err, file) => {
+        if (err) {
+          return Alerts.toast("Something went wrong", "warning");
+        }
+        if (file) {
+          this.setState({fileId: file._id});
+          const digitalInfo = {
+            fileId: file._id,
+            category: this.state.categoryValue
+          };
+
+          this.props.onProductFieldChange(this.props.product._id, "productFileId", file._id);
+          this.props.onProductFieldChange(this.props.product._id, "digitalInfo", digitalInfo);
+          return Alerts.toast("File successfully uploaded", "success");
+          }
+        return "File Insert Failed";
+      });
+    };
+    if (this.state.categoryValue === "audio") {
+      return insertDigitalFile(Audio, Alerts);
+    } else if (this.state.categoryValue === "video") {
+      return insertDigitalFile(Video, Alerts);
+    } else if (this.state.categoryValue === "book") {
+      return insertDigitalFile(Book, Alerts);
+    } else if (this.state.categoryValue === "software") {
+      return insertDigitalFile(Software, Alerts);
+    }
+    return "Invalid Category Specified";
+  }
+  get download() {
+    const result = Audio.findOne({_id: this.state.fileId});
+    return (new FS.File(result).url());
+  }
+
+  deleteFile() {
+    if (this.state.fileId) {
+      return (
+        <button className="btn btn-danger no-round">Delete Uploaded File</button>
+      );
+    }
+    return null;
+  }
+
+  renderDigitalDetails() {
+    if (this.props.hasAdminPermission && this.state.digital) {
+      return (
+        <div>
+          <Divider
+            label="Digital product options"
+          />
+          <FieldGroup
+            componentClass="select"
```

Comment:
Use the same approach @andela-jkanyua used for static pages, they are static but dynamically generated.